lib/cpp/src/thrift/transport/TBufferTransports.cpp

```diff
@@ -361,9 +361,13 @@ void TMemoryBuffer::ensureCanWrite(uint32_t len) {
   }
 
   // Grow the buffer as necessary.
-  uint32_t new_size = bufferSize_;
+  uint64_t new_size = bufferSize_;
   while (len > avail) {
     new_size = new_size > 0 ? new_size * 2 : 1;
+    if (new_size > std::numeric_limits<uint32_t>::max()) {
```

Comment:
Shouldn't this be int32_t instead of uint32_t, if we want to test against 2GB?  Sorry I didn't catch this before.